--- libtickit-9999/Makefile.orig	2020-10-01 18:55:26.000000000 +0200
+++ libtickit-9999/Makefile	2022-01-17 18:09:18.736605263 +0100
@@ -136,17 +136,6 @@
 install-man:
 	install -d $(DESTDIR)$(MAN3DIR)
 	install -d $(DESTDIR)$(MAN7DIR)
-	for F in man/*.3; do \
-	  gzip <$$F >$(DESTDIR)$(MAN3DIR)/$${F#man/}.gz; \
-	done
-	for F in man/*.7; do \
-	  gzip <$$F >$(DESTDIR)$(MAN7DIR)/$${F#man/}.gz; \
-	done
-	while read FROM EQ TO; do \
-	  if [ -n "$$FROM" ]; then \
-	    ln -sf $$TO.gz $(DESTDIR)$(MAN3DIR)/$$FROM.gz; \
-	  fi; \
-	done < man/also
 
 HTMLDIR=html
 
