diff --git a/linux/sk-prog b/linux/sk-prog
index ce9aa66..2926107 100644
--- a/linux/sk-prog
+++ b/linux/sk-prog
@@ -1,6 +1,7 @@
 partial alphanumeric_keys
 xkb_symbols "sk-prog" {
 
+    include "latin"
     name[Group1] = "Slovak - Prog";
 
     key <TLDE>	{ [     grave, asciitilde,     NoSymbol,     NoSymbol ]	};
@@ -58,4 +59,4 @@ xkb_symbols "sk-prog" {
     key <SPCE>  { [     space,      space, nobreakspace, nobreakspace ] };
 
     include "level3(ralt_switch)"
-};
\ No newline at end of file
+};
